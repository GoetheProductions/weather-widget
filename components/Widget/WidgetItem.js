const WidgetItem = ({ children }) => <li className="list-group-item">{children}</li>;

export default WidgetItem;
