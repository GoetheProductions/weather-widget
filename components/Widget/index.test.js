import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Component from './index';
import WidgetItem from './WidgetItem';

const component = renderer.create(
  <Component title="test">
    <WidgetItem />
    <WidgetItem />
  </Component>,
);

test('Widget matches snapshot', () => {
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should have more than one child', () => {
  expect(component.toTree().props.children.length).toBeGreaterThan(0);
});

test('Should have WidgetItem as child', () => {
  expect(component.toTree().props.children[0].type).toEqual(WidgetItem);
});
