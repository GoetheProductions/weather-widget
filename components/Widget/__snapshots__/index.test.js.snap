// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget matches snapshot 1`] = `
<div
  className="widget"
  style={
    Object {
      "margin": "10px",
      "width": "300px",
    }
  }
>
  <div
    className="panel panel-info"
  >
    <div
      className="panel-heading"
    >
      Weather in 
      <b>
        test
      </b>
    </div>
    <ul
      className="list-group"
    >
      <li
        className="list-group-item"
      />
      <li
        className="list-group-item"
      />
    </ul>
  </div>
</div>
`;
